src/main/java/org/opensearch/ad/task/ADTaskManager.java

```diff
@@ -1260,15 +1286,26 @@ private void cleanOldAdTaskDocs(IndexResponse response, ADTask adTask, ActionLis
                         ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.nextToken(), parser);
                         ADTask adTask = ADTask.parse(parser, searchHit.getId());
                         logger.debug("Delete old task: {} of detector: {}", adTask.getTaskId(), adTask.getDetectorId());
-                        // TODO: add deleted task in cache to delete their AD results in hourly cron job
                         bulkRequest.add(new DeleteRequest(CommonName.DETECTION_STATE_INDEX).id(adTask.getTaskId()));
                     } catch (Exception e) {
                         listener.onFailure(e);
                     }
                 }
                 client.execute(BulkAction.INSTANCE, bulkRequest, ActionListener.wrap(res -> {
                     logger.info("AD tasks deleted for detector {}", detectorId);
-                    // TODO: delete child tasks of HC detector task
+                    BulkItemResponse[] bulkItemResponses = res.getItems();
```

Comment:
I didn't find you use r.getTotal, but you enabled ".trackTotalHits(true) at line 1250".  If you don't need it, no need to turn it on as it is expensive.

Reply:
Seems some debug code, will remove it